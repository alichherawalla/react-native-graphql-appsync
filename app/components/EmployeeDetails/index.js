/**
 *
 * EmployeeDetails
 * This component is meant to show the employee details.
 * It wil show the addresses, skills and other details of an employee
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import { ScrollView, StyleSheet, View } from 'react-native'
import { List } from 'react-native-paper'
import get from 'lodash/get'
import { injectIntl } from 'react-intl'
import Container from 'app/components/Container'
import T from 'app/components/T'
import LabelledText from 'app/components/LabelledText'
import For from 'app/components/For'
import Divider from 'app/components/Divider'

import { getAppBarWithBack } from 'app/components/AppBar'
import NavigationService from 'app/services/NavigationService'

function EmployeeDetails({
  intl,
  navigation: {
    state: {
      params: { employee, handleEmployeeDelete }
    }
  }
}) {
  const addresses = get(employee, 'address.items', [])
  const skills = get(employee, 'skills.items', [])
  const renderAddresses = address => (
    <View>
      <LabelledText labelId="line_1" value={get(address, 'line1', '')} />
      <LabelledText labelId="line_2" value={get(address, 'line2', '')} />
      <LabelledText labelId="city" value={get(address, 'city', '')} />
      <LabelledText labelId="state" value={get(address, 'state', '')} />
      <LabelledText labelId="zipcode" value={get(address, 'zipcode', '')} />
      <Divider marginHorizontal={4} />
    </View>
  )

  const renderSkills = skill => (
    <View>
      <T text={get(skill, 'name', '')} />
      <Divider marginHorizontal={4} />
    </View>
  )

  const renderBackMenuWithDeleteAndEdit = () =>
    getAppBarWithBack(intl.formatMessage({ id: 'employee_details' }), [
      {
        type: 'pencil',
        onPress: () => {
          NavigationService.navigate('EditEmployeeDetailsScreen', {
            employee,
            isEditing: true
          })
        }
      },
      {
        type: 'delete',
        onPress: () => {
          if (handleEmployeeDelete) {
            handleEmployeeDelete(employee)
          }
        }
      }
    ])

  return (
    <Container testID="employee-details">
      {renderBackMenuWithDeleteAndEdit()}
      <ScrollView style={styles.container}>
        <LabelledText labelId="first_name" value={get(employee, 'firstname')} />
        <LabelledText labelId="last_name" value={get(employee, 'lastname')} />

        <Divider />
        <List.Accordion
          title={intl.formatMessage(
            { id: 'address' },
            { length: addresses.length }
          )}
        >
          <For marginLeft={20} of={addresses} renderItem={renderAddresses} />
        </List.Accordion>

        <Divider />
        <List.Accordion
          title={intl.formatMessage(
            { id: 'skills' },
            { length: skills.length }
          )}
        >
          <For marginLeft={20} of={skills} renderItem={renderSkills} />
        </List.Accordion>
      </ScrollView>
    </Container>
  )
}

const styles = StyleSheet.create({
  container: {
    margin: 12
  }
})
EmployeeDetails.propTypes = {
  intl: PropTypes.object,
  navigation: PropTypes.shape({
    state: {
      params: {
        employee: PropTypes.shape({
          id: PropTypes.string,
          firstname: PropTypes.string,
          lastname: PropTypes.string,
          address: PropTypes.arrayOf(
            PropTypes.shape({
              line1: PropTypes.string,
              line2: PropTypes.string,
              city: PropTypes.string,
              state: PropTypes.string,
              zipcode: PropTypes.string
            })
          ),
          skills: PropTypes.arrayOf(
            PropTypes.shape({
              id: PropTypes.string,
              name: PropTypes.string
            })
          )
        })
      }
    }
  }),
  handleEmployeeDelete: PropTypes.func
}

EmployeeDetails.defaultProps = {
  handleEmployeeDelete: () => {}
}

export default injectIntl(EmployeeDetails)
