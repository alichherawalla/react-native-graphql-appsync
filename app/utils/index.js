import { fonts } from 'app/themes'

export const getRandomKey = () =>
  Math.random()
    .toString(36)
    .substr(2, 3)
export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms))

export const rnFontSizeAdapter = type => ({
  fontSize: parseInt(
    fonts.size[type]()[0]
      .replace('font-size: ', '')
      .replace('px;', '')
      .trim(),
    10
  )
})

export const rnFontWeightAdapter = type => ({
  fontSize: parseInt(
    fonts.weights[type]()[0]
      .replace('font-weight: ', '')
      .trim(),
    10
  )
})

export const NEW_EMPLOYEE = {
  employee: {
    address: [{}],
    skills: [{}]
  }
}

export const SIGNUP_CONFIG = {
  header: 'Sign up to buildops',
  hideAllDefaults: true,
  defaultCountryCode: '1',
  signUpFields: [
    {
      label: 'User name',
      key: 'username',
      required: true,
      displayOrder: 1,
      type: 'string'
    },
    {
      label: 'Password',
      key: 'password',
      required: true,
      displayOrder: 2,
      type: 'password'
    },
    {
      label: 'Email',
      key: 'email',
      required: true,
      displayOrder: 4,
      type: 'string'
    }
  ]
}
