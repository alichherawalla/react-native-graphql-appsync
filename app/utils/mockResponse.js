import uuid from 'uuid'
export const employeeData = [
  {
    id: uuid(),
    firstname: 'MACTESTER',
    lastname: 'Doe'
  },
  {
    id: 2,
    firstname: 'Jonathan',
    lastname: 'Doe',
    address: [
      {
        line1: '3022 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3024 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3026 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3028 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [{ id: 6, name: 'Senior Electrician' }]
  },
  {
    id: 3,
    firstname: 'Steve',
    lastname: 'Doe',
    address: [
      {
        line1: '3028 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 4, name: 'Senior Technician' },
      { id: 5, name: 'Senior Plumber' }
    ]
  },
  {
    id: 4,
    firstname: 'Nathan',
    lastname: 'Doe',
    address: [
      {
        line1: '3022 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3024 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3026 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 4, name: 'Senior Technician' },
      { id: 5, name: 'Senior Plumber' },
      { id: 6, name: 'Senior Electrician' }
    ]
  },
  {
    id: 5,
    firstname: 'Albert',
    lastname: 'Doe',
    address: [
      {
        line1: '3022 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3024 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3026 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3028 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 4, name: 'Senior Technician' },
      { id: 5, name: 'Senior Plumber' },
      { id: 6, name: 'Senior Electrician' }
    ]
  }
]
