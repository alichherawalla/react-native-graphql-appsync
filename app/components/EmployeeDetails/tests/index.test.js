/**
 *
 * Tests for EmployeeDetails
 *
 */

import React from 'react'
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from 'app/utils/testUtils'
import EmployeeDetails from '../index'
import { employeeData } from '../../../utils/mockResponse'

describe('<EmployeeDetails />', () => {
  const props = {}
  beforeEach(() => {
    props.navigation = {
      state: {
        params: { employee: employeeData()[0] }
      }
    }
  })
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<EmployeeDetails {...props} />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should contain 1 EmployeeDetails component', () => {
    const { getAllByTestId } = renderWithIntl(<EmployeeDetails {...props} />)
    expect(getAllByTestId('employee-details').length).toBe(1)
  })
})
