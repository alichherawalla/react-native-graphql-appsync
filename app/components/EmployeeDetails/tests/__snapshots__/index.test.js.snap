// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployeeDetails /> should render and match the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
      testID="employee-details"
    >
      <SafeAreaView
        style={
          Array [
            Object {
              "backgroundColor": "#6200ee",
              "elevation": 4,
              "zIndex": 0,
            },
            Object {
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#6200ee",
              "display": "flex",
              "elevation": 0,
              "flexDirection": "row",
              "height": 56,
              "marginTop": 0,
              "paddingHorizontal": 4,
            }
          }
          testID="app-bar"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "marginLeft": 0,
                }
              }
              testID="for"
            >
              <TouchableHighlight
                accessibilityComponentType="button"
                accessibilityRole="button"
                accessibilityStates={Array []}
                accessibilityTraits="button"
                activeOpacity={0.85}
                centered={true}
                delayPressOut={100}
                disabled={false}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "margin": 6,
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 18,
                        "height": 36,
                        "width": 36,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
                underlayColor="rgba(255, 255, 255, 0.16)"
              >
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                        },
                        Object {
                          "height": 24,
                          "width": 24,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "opacity": 1,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "transform": Array [
                            Object {
                              "rotate": "0deg",
                            },
                          ],
                        }
                      }
                    >
                      <Text
                        accessibilityElementsHidden={true}
                        allowFontScaling={false}
                        importantForAccessibility="no-hide-descendants"
                        pointerEvents="none"
                        style={
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 24,
                            },
                            Array [
                              Object {
                                "transform": Array [
                                  Object {
                                    "scaleX": 1,
                                  },
                                ],
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                            ],
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        ÔÅç
                      </Text>
                    </View>
                  </View>
                </View>
              </TouchableHighlight>
            </View>
            <TouchableWithoutFeedback>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "paddingHorizontal": 12,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  accessibilityRole="header"
                  accessibilityTraits="header"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "color": "#ffffff",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "fontSize": 17,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Employee Details
                </Text>
              </View>
            </TouchableWithoutFeedback>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginLeft": 0,
              }
            }
            testID="for"
          >
            <TouchableHighlight
              accessibilityComponentType="button"
              accessibilityRole="button"
              accessibilityStates={Array []}
              accessibilityTraits="button"
              activeOpacity={0.85}
              centered={true}
              delayPressOut={100}
              disabled={false}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
              underlayColor="rgba(255, 255, 255, 0.16)"
            >
              <View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "transform": Array [
                          Object {
                            "rotate": "0deg",
                          },
                        ],
                      }
                    }
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 24,
                          },
                          Array [
                            Object {
                              "transform": Array [
                                Object {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ],
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      Ôè´
                    </Text>
                  </View>
                </View>
              </View>
            </TouchableHighlight>
            <TouchableHighlight
              accessibilityComponentType="button"
              accessibilityRole="button"
              accessibilityStates={Array []}
              accessibilityTraits="button"
              activeOpacity={0.85}
              centered={true}
              delayPressOut={100}
              disabled={false}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
              underlayColor="rgba(255, 255, 255, 0.16)"
            >
              <View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "transform": Array [
                          Object {
                            "rotate": "0deg",
                          },
                        ],
                      }
                    }
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 24,
                          },
                          Array [
                            Object {
                              "transform": Array [
                                Object {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ],
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ÔáÄ
                    </Text>
                  </View>
                </View>
              </View>
            </TouchableHighlight>
          </View>
        </View>
      </SafeAreaView>
      <ScrollView
        style={
          Object {
            "margin": 12,
          }
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 4,
              "marginVertical": 4,
            }
          }
          testID="labelled-text"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Object {
                  "color": undefined,
                  "fontSize": 15,
                  "fontWeight": "normal",
                },
              ]
            }
            testID="t"
          >
            First Name
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Object {
                  "color": "#808080",
                  "fontSize": 14,
                  "fontWeight": "normal",
                },
              ]
            }
            testID="t"
          >
            MACTESTER
          </Text>
        </View>
        <View
          style={
            Object {
              "marginHorizontal": 4,
              "marginVertical": 4,
            }
          }
          testID="labelled-text"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Object {
                  "color": undefined,
                  "fontSize": 15,
                  "fontWeight": "normal",
                },
              ]
            }
            testID="t"
          >
            Last Name
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Object {
                  "color": "#808080",
                  "fontSize": 14,
                  "fontWeight": "normal",
                },
              ]
            }
            testID="t"
          >
            Doe
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              Object {
                "marginHorizontal": 0,
                "marginVertical": 10,
              },
            ]
          }
          testID="divider"
        />
        <View>
          <TouchableHighlight
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityTraits="button"
            activeOpacity={0.85}
            delayPressOut={100}
            disabled={false}
            style={
              Array [
                false,
                Array [
                  Object {
                    "padding": 8,
                  },
                  undefined,
                ],
              ]
            }
            underlayColor="rgba(0, 0, 0, 0.1)"
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "margin": 8,
                    },
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Addresses (0)
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "margin": 8,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "rgba(0, 0, 0, 0.87)",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ÔÖÄ
                </Text>
              </View>
            </View>
          </TouchableHighlight>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              Object {
                "marginHorizontal": 0,
                "marginVertical": 10,
              },
            ]
          }
          testID="divider"
        />
        <View>
          <TouchableHighlight
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityTraits="button"
            activeOpacity={0.85}
            delayPressOut={100}
            disabled={false}
            style={
              Array [
                false,
                Array [
                  Object {
                    "padding": 8,
                  },
                  undefined,
                ],
              ]
            }
            underlayColor="rgba(0, 0, 0, 0.1)"
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "margin": 8,
                    },
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Skills (0)
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "margin": 8,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "rgba(0, 0, 0, 0.87)",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ÔÖÄ
                </Text>
              </View>
            </View>
          </TouchableHighlight>
        </View>
      </ScrollView>
    </SafeAreaView>
  </View>
</View>
`;
